proto/uber/cadence/api/v1/common.proto

```diff
@@ -124,4 +124,27 @@ message IsolationGroupPartition {
 
 message IsolationGroupConfiguration {
     repeated IsolationGroupPartition isolation_groups = 1;
-}
\ No newline at end of file
+}
+
+message AsyncWorkflowConfiguration {
+  // predefined_queue_name is the name of the predefined queue in cadence server config's asyncWorkflowQueues
+  string predefined_queue_name = 1;
+
+  // Below fields are only used if predefined_queue_name is not set
+  AsyncWorkflowQueueType queue_type = 2;
+  AsyncWorkflowKafkaQueueConfiguration kafka_config = 3;
```

Comment:
Should it be moved into a oneof block?

Reply:
Make sense, updated. I missed because I started with writing thrift struct and copied over to proto. Thrift doesn't have such concept and looks like we just declare every field as optional. See HistoryEvent for example.